src/AutoRest.CSharp/MgmtTest/Extensions/CodeWriterExtensions.cs

```diff
@@ -151,8 +152,18 @@ private static CodeWriter AppendComplexFrameworkTypeValue(this CodeWriter writer
             writer.Append($"new {type}(");
             foreach (var parameter in publicCtor.GetParameters())
             {
-                // we here assume the parameter name is the same as the serialized name of the property. This is not 100% solid
-                var value = exampleValue.Properties[parameter.Name!];
+                ExampleValue value;
+                if (exampleValue.Properties.ContainsKey(parameter.Name!))
+                    // we here assume the parameter name is the same as the serialized name of the property. This is not 100% solid
+                    value = exampleValue.Properties[parameter.Name!];
+                else
+                {
+                    // if we can't find it directly, try to figure out the serailizedName form propertyMetadataDict
+                    string? serializedName = propertyMetadataDict.FirstOrDefault(p => string.Equals(p.Key, parameter.Name!, StringComparison.OrdinalIgnoreCase)).Value?.SerializedName;
+                    if (string.IsNullOrEmpty(serializedName) || !exampleValue.Properties.ContainsKey(serializedName))
+                        throw new InvalidOperationException($"Unabled to find example value for ctor parameter, type={type.FullName}, param={parameter.Name ?? "<null>"}");
+                    value = exampleValue.Properties[serializedName];
+                }
```

Comment:
How about we just simplify this by using `TryGetValue`
```suggestion
                if (!exampleValue.Properties.TryGetValue(parameter.Name!, out var value))
                {
                    // if we can't find it directly, try to figure out the serailizedName form propertyMetadataDict
                    string? serializedName = propertyMetadataDict.FirstOrDefault(p => string.Equals(p.Key, parameter.Name!, StringComparison.OrdinalIgnoreCase)).Value?.SerializedName;
                    if (string.IsNullOrEmpty(serializedName) || !exampleValue.Properties.ContainsKey(serializedName))
                        throw new InvalidOperationException($"Unabled to find example value for ctor parameter, type={type.FullName}, param={parameter.Name ?? "<null>"}");
                    value = exampleValue.Properties[serializedName];
                }
```

Reply:
Also could we update the description to show in which real life case, would this KeyNotFoundException happen?
I kind of do not get why this could possibly happen and this is also the reason when I wrote this piece of code I did not add any extra checks.

Please keep in mind, if there is something wrong in the swagger example, we should not mitigate that here, instead we should open a PR in the swagger repo to fix the example json files.